src/components/RouteWithProps/index.jsx

```diff
@@ -24,7 +26,9 @@ export default class RouteWithProps extends Component {
         strict={strict}
         location={location}
         sensitive={sensitive}
-        render={renderProps => <Component {...renderProps} {...props} />}
+        render={({ staticContext, ...renderProps }) => (
```

Comment:
I missed this on the first round -- what is `staticContext` and why does it need to be filtered out here?

Reply:
A `staticContext` property is inserted whenever we use react-router. If we don't remove it, it will end up as a prop to another compoent that does not recognize this property and an error will be thrown in the console: ``` "React does not recognize the `staticContext` prop on a DOM element"  ```.

Reply:
Ah, OK :)